source/logrepl/internal/subscription.go

```diff
@@ -87,14 +91,25 @@ func CreateSubscription(
 			Msgf("replication slot %q already exists", slotName)
 	}
 
+	// If a ConsistentPoint is available use it, otherwise fall back to the startLSN
+	slotLSN, err = pglogrepl.ParseLSN(result.ConsistentPoint)
+	if err != nil {
+		if startLSN == 0 {
+			return nil, fmt.Errorf("failed to parse consistent wal lsn: %w", err)
+		}
```

Comment:
Wait, won't this prevent me to attach a new pipeline to an existing replication slot? 😕 What if we query [`pg_replication_slots.restart_lsn`](https://www.postgresql.org/docs/9.4/catalog-pg-replication-slots.html) and use that as our start LSN (in case no position was provided and we didn't create the slot)?

Reply:
Yes. I wanted to do this work as a follow up, because I’d like to refactor what type of connection the subscription is using. Rather than using raw pgconn, it will be best to use pgx so we can take advantage of the querier interface. 

The idea is to not release new version until this is addressed, which will close another issue with invalid LSN in position.